aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -283,7 +283,7 @@ std::pair<const AnnotatedKernel&, const char*> OperatorEntry::computeDispatchTab
   }
 
   // 3. Backend fallback
-  auto dispatch_ix = static_cast<uint8_t>(dispatch_key);
+  auto dispatch_ix = DispatchKeySet(dispatch_key).getDispatchTableIndexForDispatchKeySet();
```

Comment:
Is it necessary to construct the entire DispatchKeySet to get the index into the dispatch table?

Reply:
Also nit: this operation: `DispatchKeySet(dispatch_key).getDispatchTableIndexForDispatchKeySet()` happens a lot. Is there a reason why the old `getDispatchTableIndexForDispatchKey` couldn't have been updated to just do `DispatchKeySet(dispatch_key).getDispatchTableIndexForDispatchKeySet()`?

Reply:
> Is it necessary to construct the entire DispatchKeySet to get the index into the dispatch table?

You're right that this is indirect (and potentially slower): we're going from key -> keyset -> index. I thought about writing a separate function that goes directly from key -> index, but the downside is we'd end up with two functions for calculating the index need to be kept in sync (and could cause subtle bugs if one is incorrect). The perf impact of that indirection is also probably fine since this is all at registration time.

> Also nit: this operation: DispatchKeySet(dispatch_key).getDispatchTableIndexForDispatchKeySet() happens a lot. Is there a reason why the old getDispatchTableIndexForDispatchKey couldn't have been updated to just do DispatchKeySet(dispatch_key).getDispatchTableIndexForDispatchKeySet()?

Good point. I'll add back `getDispatchTableIndexForDispatchKey()` as a helper that just does this.